app/Http/Controllers/Admin/Admin.php

```diff
@@ -180,41 +283,104 @@ public function deleteClass($id)
     }
     public function infoTeacher($id)
     {
-        $teacher = Teacher::find($id);
-        $classes = DB::table('classes')->where('teacher_id', '=', $id)->get();
+        $teacher = Teacher::findOrFail($id);
+        $classes = DB::table('classes')->where('teacher_id', '=', $id)
+            ->join('subjects','subjects.id','=','classes.subject_id')
+            ->select('classes.*','subjects.name')->get();
         return view('admin.teacher.info', ['teacher' => $teacher,
                                                  'classes' => $classes,]);
     }
-
+    public function infoSubject($id)
+    {
+        $subject = Subject::findOrFail($id);
+        $classes =  $subject->classes()->with('students')->with('teacher')->get();
+        return view('admin.subjects.info',[
+            'subject' => $subject,
+            'classes' => $classes,
+        ]);
+    }
+    public function infoClass($id)
+    {
+        $class = Classes::findOrFail($id);
+        $students = $class->students()->orderBy('students.name')->get();
+        return view('admin.class.info', ['class' => $class,
+            'students' => $students,
+            'i' => 0,
+        ]);
+    }
     public function findAvg($id, Request $request)
     {
         $student = Student::where('id','=',$id)->first();
-        $classStudent = DB::table('student_class')->join('classes', 'student_class.class_id', '=', 'classes.id')
-            ->where('student_id', '=', $id)->where('semester', '=', $request->semester)->get();
+        $classes = $student->classes()->where('semester','=',$request->semester)
+            ->with('teacher')->with('subject')->get();
         $i = 0;
         $avg = 0;
-        foreach($classStudent as $class)
+        foreach($classes as $class)
         {
             $i++;
-            if($class->score >= 0)
-                $avg += $class->score;
+            if($class->pivot->score >= 0)
+                $avg += $class->pivot->score;
         }
         if($i == 0)
         {
             return back()->with('danger', 'Semester is not exist!!');
         }
         $avg = $avg/$i;
-        return view('admin.student.info')->with(['classStudents' => $classStudent,
+        return view('admin.student.info')->with(['classStudents' => $classes,
                              'avg' => $avg,
                              'success' => 'Successfully',
                                 'student' => $student,
                             ]);
     }
-    //
+
+    public function searchStudent(Request $request)
+    {
+        $student = Student::where('id', 'like', '%'.$request->nameOrId.'%')
+            ->orWhere('name', 'like', '%'.$request->nameOrId.'%')->orWhere('class', 'like', '%'.$request->nameOrId.'%')
+            ->select('students.*')->paginate(5);
+        if($student && $student->count() >0){
+            return view('admin.student.list', ['data' => $student]);
+        }
```

Comment:
- Mặc định là lây tất cả thì cần gì phải `select`
- Nếu muốn kiểm tra số lượng bản ghi lấy được thì kiểm tra bên view ý

Reply:
Em thừa cái select đó. Xử lý bên view là sao hả anh. Sao lại xử lý bên view() nên dùng controller chứ nhỉ. mà cái này em kiểm tra có hay ko để còn trả về message mà

Reply:
Bên view mình kiểm tra `@if( $student->count() >0)` thì in ra kết quả tìm kiếm còn `@else` thì in ra mesage là không có

Reply:
em nghĩ là cái thao tác tính toán nào mà thực hiện được ở controller thì làm ở controller chứ. Sao cái này lại làm ở view được. ví dụ em giả sử em muốn redirect sang một route khác thì sao.